src/readers/reader_tum.py

```diff
@@ -0,0 +1,129 @@
+import cv2
+import mrob
+import numpy as np
+import open3d as o3d
+import os
+
+from nptyping import Float, NDArray, Shape, UInt8
+from pathlib import Path
+from reader import Reader
+from src.core import Database, Image, PointCloud
+
+
+class ReaderTUM(Reader):
+    def __init__(
+        self,
+        path_to_dataset: str,
+        intrinsics: NDArray[Shape["3, 3"], Float],
+        dist_coeff: NDArray[Shape["5"], Float],
+        max_difference: int = 1000,
+    ):
+        self.path_to_dataset = path_to_dataset
+        self.max_difference = max_difference
+        self.intrinsics = intrinsics
+        self.dist_coeff = dist_coeff
+        self._shape = 640, 480
+        self._depth_scale = 5000
+
+    def read_dataset(self) -> Database:
+        return super().read_dataset()
+
+    def _get_images_pcds_traj(
+        self,
+    ) -> tuple[list[Image], list[PointCloud], list[NDArray[Shape["4, 4"], Float]]]:
+        depth_with_timestamps = self.__read_folder(
+            os.path.join(self.path_to_dataset, "depth")
+        )
+        rgb_with_timestamps = self.__read_folder(
+            os.path.join(self.path_to_dataset, "rgb")
+        )
+        matches = self.__associate(rgb_with_timestamps, depth_with_timestamps)
+        timestamps, traj = self.__read_trajectory()
+        timestamps = np.asarray(timestamps)
+        images = []
+        pcds = []
+        res_traj = []
+        for (rgb_timestamp, depth_timestamp) in matches:
+            diff = abs(timestamps - np.mean([rgb_timestamp, depth_timestamp]))
```

Comment:
Let's move association logic to separate function with proper naming, could not clear for new users